src/app/setup-wizard/pre-visit/pre-assessment-setup/pre-assessment-setup.component.ts

```diff
@@ -111,21 +128,28 @@ export class PreAssessmentSetupComponent {
     let utilityTypes = AssessmentOptions.find(
       _assessmentOption => _assessmentOption.assessmentType == assessment.assessmentType)?.utilityTypes || [];
     assessment.utilityTypes = utilityTypes; // track all utility types
-    // utility type is not the default if the assessment type changes
-    if (assessment.utilityType !== utilityTypes[0]) {
-      assessment.utilityType = utilityTypes?.[0]; // update to the first/default utility type
-      await this.utilityTypeChange(assessment);
-    } else {
-      await this.saveChanges(assessment);
-    }
+    await this.calculateEnergyUse(assessment);
   }
 
-  async utilityTypeChange(assessment: IdbAssessment) {
-    let _energyDefaultUnit = UtilityOptions.find(
-      _utilityOption => _utilityOption.utilityType == assessment.utilityType)?.energyDefaultUnit;
-    if (assessment.unitOptionValue !== _energyDefaultUnit.value) {
-      assessment.unitOptionValue = _energyDefaultUnit.value;
-    }
+  getCompany(assessment: IdbAssessment) {
+    // get company data for energy unit conversion
+    return this.companies.find(_company => _company.guid == assessment.companyId);
+  }
+
+  async calculateEnergyUse(assessment: IdbAssessment) {
+    let result = 0;
+    assessment.utilityTypes.forEach(utilityType => {
+      let utilityEnergyUse: UtilityEnergyUse = assessment.utilityEnergyUses.find(
+        _energyUse => _energyUse.utilityType == utilityType);
+      if (utilityEnergyUse.include) {
+        let convertedValue = this.convertValue.convertValue(
+          utilityEnergyUse.energyUse,
+          utilityEnergyUse.unit,
+          this.getCompany(assessment).companyEnergyUnit).convertedValue;
```

Comment:
instead of adding the `getCompany` logic and the subscription for companies. Here you can just do:

`this.companyIdbService.getByGUID(assessment.companyId)`

Reply:
Thanks! Update with this method to pass the value to a component variable at onInit, which the html template also needs to display on the UI